mat/dense_example_test.go

```diff
@@ -128,3 +128,89 @@ func ExampleDense_Inverse() {
 	//     ⎣  0  0.5⎦
 	//
 }
+
+func ExampleDense_Mul() {
+	// Initialize two matrices, a and b.
+	a := mat.NewDense(2, 2, []float64{4, 0, 0, 4})
```

Comment:
I think we should go through the examples and change the linear representation of the `[]float64` to something that represents the matrix layout:

```
a := mat.NewDense(2, 2, []float64{
    4, 0,
    0, 4,
})
```

This can happen in another PR, since there are others that look the same already.

Reply:
seems reasonable, it makes it easier to visualize the matrix at a glance.